### PR TITLE
Simplifies Start button logic in lobby

### PR DESCRIPTION
**Pull request by deniz1a**
This does the same thing but is simpler.

### CODE REVIEW

**Comment by Unit158** [comment]
:-1: this still doesn't fix the issue.

File: OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs [hunk 1]
```diff
@@ -297,15 +297,8 @@ void CloseWindow()
...
-					{
-						var cl = orderManager.LobbyInfo.ClientInSlot(sl.Key);
-
-						// Bots and admins don't count
-						return cl != null && !cl.IsAdmin && cl.Bot == null && !cl.IsReady;
-					};
-
-					if (orderManager.LobbyInfo.Slots.Any(notReady))
+					// Bots and admins don't count
+					if (orderManager.LobbyInfo.Clients.Any(c => c.Slot != null && !c.IsAdmin && c.Bot == null && !c.IsReady))
```
Comment (pchote) [inline OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs]: This doesn't do the same thing.  The old code excluded observers, which don't have a slot associated with them.

Comment (deniz1a) [inline OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs]: What about ```c => c.Slot != null```?

**Comment by abcdefg30** [comment]
Looks good. :+1:

**Comment by RoosterDragon** [comment]
:+1: Spectators are handled correctly.

**Pull request merged**